osutil/disks/disks.go

```diff
@@ -216,8 +220,15 @@ var (
 	_ = error(PartitionNotFoundError{})
 )
 
+type deviceMapperBackResolversOpts struct {
+	dmUUID []byte
+	dmName []byte
+	// this is needed by the verity device resolver
+	idFsUUID string
```

Comment:
is this reliable? eg. for a squashfs image set up with veritysetup I only see the following properties:
```
maciek@galeon:~/work/canonical udevadm info /dev/mapper/test-device                      
P: /devices/virtual/block/dm-1
M: dm-1
R: 1
U: block
T: disk
D: b 254:1
N: dm-1
L: 0
S: disk/by-id/dm-name-test-device
S: disk/by-id/dm-uuid-CRYPT-VERITY-ede12854d14b4fa1bd9eff6a159a8e2d-test-device
S: mapper/test-device
Q: 135
E: DEVPATH=/devices/virtual/block/dm-1
E: DEVNAME=/dev/dm-1
E: DEVTYPE=disk
E: DISKSEQ=135
E: MAJOR=254
E: MINOR=1
E: SUBSYSTEM=block
E: USEC_INITIALIZED=126283443149
E: DM_UDEV_DISABLE_LIBRARY_FALLBACK_FLAG=1
E: DM_UDEV_PRIMARY_SOURCE_FLAG=1
E: DM_ACTIVATION=1
E: DM_NAME=test-device
E: DM_UUID=CRYPT-VERITY-ede12854d14b4fa1bd9eff6a159a8e2d-test-device
E: DM_UDEV_RULES_VSN=3
E: ID_FS_VERSION=4.0
E: ID_FS_BLOCKSIZE=131072
E: ID_FS_SIZE=9558421
E: ID_FS_TYPE=squashfs
E: ID_FS_USAGE=filesystem
E: NVME_HOST_IFACE=none
E: DEVLINKS=/dev/disk/by-id/dm-name-test-device /dev/disk/by-id/dm-uuid-CRYPT-VERITY-ede12854d14b4fa1bd9eff6a159a8e2d-test-device /dev/mapper/test-device
E: TAGS=:systemd:
E: CURRENT_TAGS=:systemd:
```

Reply:
So I hadn't considered the squashfs use case. This API is called today by:
- `PartitionUUIDFromMountPoint()` which calls it for encrypted devices.
- `MountPointIsFromDisk()` which is the one relevant here. This is called in various places by `snap-bootstrap` to verify that mountpoints that correspond to physical partitions come from the same disk as another partition. 

Do you have a specific use case in mind for making this work for mountpoints for squashfs archives? 

- If not, we could add a safety check in the new resolver (as you suggested in the previous comment) for this to be skipped either if `ID_FS_UUID` is missing or maybe do some filtering based on the `ID_FS_TYPE`.
- If yes some extra logic will be required as I haven't figured a way to map, for a squashfs type (like the one you commented above), its mountpoint to the underlying data partition based on udev information. The `DM_UUID` field cannot be used as in the luks2 case since this one points to the uuid of the dm-verity hash device and not the data device that (I think) we would need.

Reply:
Why not using `DM_UUID` like we do with dm-crypt?

Reply:
I really do not think depending on any `ID_FS_*` is right. If that unblocks you, this is fine, we can merge it. But it will need a follow up, and identify dm-verity devices in udev using libcryptsetup. We should also clean up the dm-crypt resolver as well.

Reply:
Sounds like a good idea then. Let's look into this in a followup.

Reply:
Can we please add a TODO that captures the idea, links to a card that tracks this, before merging this?